Automattic/woocommerce-payments
# Add basic tracking for deposits card

Fixes #5881 

#### Changes proposed in this Pull Request

This PR adds tracks events for the key interactions in the deposits card (shown in `Payments > Overview`). 

- View card: `wcadmin_page_view path=payments_overview` (existing event)
- `wcadmin_wcpay_overview_deposits_view_history_click` new!
- `wcadmin_wcpay_overview_deposits_change_schedule_click` new!

![Screenshot 2023-05-02 at 3 41 34 PM](https://user-images.githubusercontent.com/4167300/235574740-514874c0-57fb-4a6d-84cc-054ef057068f.png)


#### Testing instructions

Ensure usage tracking is enabled:  `WooCommerce > Settings > Advanced > WooCommerce.com > Usage tracking`. See below for instructions on viewing tracks events.

- Ensure your store is onboarded, taking payments, and has a balance to deposit (so the card will render).
- Go to `Payments > Overview`.
  - Confirm view event fires: `wcadmin_page_view path=payments_overview`
- Go to `Payments > Overview`. Click `View full deposits history`. 
  - Confirm event fires: `wcadmin_wcpay_overview_deposits_view_history_click`.
- Go to `Payments > Overview`. Click `Change deposit schedule`. 
  - Confirm event fires: `wcadmin_wcpay_overview_deposits_change_schedule_click`.

Explore the card UI and ensure events are fired all for relevant states and actions. 

#### How to view tracks events
These instructions are for Google Chrome, might be different in other browsers.

- Open browser `Developer tools > Network` tab so you can see network requests.
- Search for `t.gif` (this is the endpoint for tracks events).
- Each request is an event.
- Click the request and view `Payload` to see params.
  - `_en` is the event name.

-------------------

- [x] Tested on mobile (or does not apply) - n/a, not a UI change

**Post merge**



- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : n/a, usage tracking does not need QA testing
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable. n/a, usage tracking does not change critical flows
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. n/a, internal-facing feature only
